tests/MongoDB.Driver.Legacy.Tests/Builders/IndexOptionsBuilderTypedTests.cs

```diff
@@ -173,5 +179,19 @@ public void TestMultipleWeights()
             string expected = "{ \"weights\" : { \"textfield\" : 2, \"idioma\" : 10 } }";
             Assert.Equal(expected, options.ToJson());
         }
+
+        [Fact]
+        public void WildcardProjection_should_return_expected_result()
+        {
+            var options = IndexOptions<TestClass>.SetWildcardProjection(x => x.textfield, true);
+            BsonDocument expected = BsonDocument.Parse("{ \"wildcardProjection\" : { \"textfield\" : 1 } }");
+            options.ToBsonDocument().Should().Be(expected);
+
+            options = IndexOptions<TestClass>
+                .SetWildcardProjection(x => x.textfield, false)
+                .SetWildcardProjection(x => x.details.text, true);
+            expected = BsonDocument.Parse("{ \"wildcardProjection\" : { \"textfield\" : 0, \"details.text\" : 1 } }");
+            options.ToBsonDocument().Should().Be(expected);
+        }
     }
 }
```

Comment:
Minor suggestion: I like adding new lines to the end of the files that we update to make the "no new line at end of file" warning go away, per the [POSIX standard](https://thoughtbot.com/blog/no-newline-at-end-of-file).

Reply:
Done